### PR TITLE
feat: add transaction fee design

### PR DESCRIPTION
**Pull request by shahin-hq**
Closes: https://app.clickup.com/t/86drqxqjr
Design: https://www.figma.com/file/34IUtRCOxp2RqPh1IVoLFx/Connect-Wallet?node-id=531%3A23550&mode=dev

This PR adds transaction fee design

Please consider this while reviewing: https://helloardent.slack.com/archives/C069X9DHWUR/p1711012690634359

### CODE REVIEW

**Comment by ItsANameToo** [comment]
leaving this open until the fee handling is available in the extension as well, otherwise this will block releases of demo

File: src/domains/transactions/components/SendModal/SendModal.blocks.tsx [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+          step="0.00000001"
+          type="number"
+          {...feeInputProperties}
+          className="border-theme-gray-400 rounded-l-lg px-3 text-md leading-5 block w-full py-2.5 focus:ring-1 ring-theme-gray-400 rounded-e-none [appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none"
+        />
+        <div className="flex flex-col border border-l-0 dark:border-theme-gray-500 overflow-hidden rounded-e-lg border-theme-gray-400 w-10 shrink-0">
+          <button
+            type="button"
+            className="flex items-center justify-center hover:bg-theme-gray-50 dark:hover:bg-theme-gray-600 basis-1/2 w-full focus:ring-gray-100 focus:ring-2 focus:outline-none relative after:content-[''] after:absolute after:border-t after:bottom-0 after:border-theme-gray-400 after:w-full"
+          >
```
[the remark this answers is not in the document]
Comment (shahin-hq) [reply src/domains/transactions/components/SendModal/SendModal.blocks.tsx:106]: Yeah, that should be handled in a specific task. I didn't want to make a gigantic PR :d

File: src/domains/transactions/components/SendModal/SendModal.tsx [hunk 2]
```diff
@@ -74,6 +81,38 @@ export const SendModal = ({
   // @TODO: is this the best way to get the coin name?
   const coin = wallet.network === NetworkType.DEVNET ? "DARK" : "ARK";
 
+  const [feeInputProperties, setFeeInputProperties] = useState(
+    {} as UseFormRegisterReturn,
+  );
+
+  useEffect(() => {
+    const inputProperties = register("fee", {
+      required: t("FEE"),
```
[the remark this answers is not in the document]
Comment (shahin-hq) [reply src/domains/transactions/components/SendModal/SendModal.tsx:90]: Ah, yeah, sure, will send a fix

[the remark this answers is not in the document]
Comment (shahin-hq) [reply src/domains/transactions/components/SendModal/SendModal.tsx]: Fixed, please check

**Review by samharperpittam — APPROVED** [review]
Approved but have added a list of known issues and a couple of new ones here to be worked on in separate PR's

https://app.clickup.com/t/86dt3c7y8

Nothing scary though.

**Review by ItsANameToo — APPROVED** [review]

File: src/domains/transactions/components/SendModal/SendModal.blocks.tsx [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+      <span className="flex flex-row sm:flex-col justify-between flex-1 items-center sm:pb-2 text-theme-gray-500 dark:text-theme-gray-200">
+        <span
+          className={cn("font-medium sm:pb-1 leading-5", {
+            "text-black dark:text-theme-gray-200": isSelected,
+          })}
+        >
+          {" "}
+          {title}{" "}
+        </span>
+        <span className="leading-4.5"> {cryptoAmount} ARK </span>
```
Comment (ItsANameToo) [inline src/domains/transactions/components/SendModal/SendModal.blocks.tsx:208]: this should be based on the network token, and not hardcoded to ARK

Comment (ItsANameToo) [reply src/domains/transactions/components/SendModal/SendModal.blocks.tsx:208]: https://app.clickup.com/t/86dt3cekb

**Pull request merged**